xblock/fields.py

```diff
@@ -780,11 +780,19 @@ class String(JSONField):
     MUTABLE = False
 
     def from_json(self, value):
-        if value is None or isinstance(value, basestring):
+        if value is None:
+            return ''
+        elif isinstance(value, basestring):
             return value
         else:
             raise TypeError('Value stored in a String must be None or a string, found %s' % type(value))
 
+    def to_json(self, value):
+        if value is None:
+            return ''
+        self._warn_deprecated_outside_JSONField()
+        return value
```

Comment:
That said, I can't think of a better option, really. We're encoding `str|None` into `str`, so we're going to lose information somewhere, no matter what we do.